source/agora/common/Deserializer.d

```diff
@@ -119,6 +119,46 @@ unittest
         Amount(100), [1, 2, 3]));
 }
 
+/// Default upper bound for deserializeLength & co
+public enum DefaultLengthUpperBound = 0x09D0;
+
+/*******************************************************************************
+
+    Deserialize a length, checks if it's within reasonable bound, and return it
+
+    This is a convenience function that calls `deserializeFull`, and checks
+    the result matches a certain bound.
+
+    It is used everywhere an array is deserialized, for example to prevent
+    a DoS if we received crafted binary data that would lead us to allocates
+    large amounts of memory, either exceeding our available memory,
+    or overloading the GC.
+
+    Params:
+        dg         = Delegate to read binary data for deserialization
+        upperBound = The value of the upper bound (inclusive).
+                     Default to 0x09D0, which is a bit less than a page on most
```

Comment:
I would move this comment to the `DefaultLengthUpperBound` declaration, because that's where the magic number appears.